### DISCUSSION TITLE
Low-level struct improvements in C# 11

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#25] [General]
# Low-level struct improvements in C# 11

C# 11 brings some major improvements for structs. You remember Span<T>, which you couldn't create?
Now you can. C# 11 allows ref fields in a struct. Why this is a cool thing and what you can do with it, you will see in a minute.

https://steven-giesel.com/blogPost/2a807220-166a-4fd0-99cc-a56765767fbd

**Comment by dadhi** [comment]
I was not able to make `StackLinkedListNode` example work in the RTM of .NET 7 and C# 11. 
Anything I miss?

**Reply by linkdotnet** [reply]
Hey @dadhi thanks for the input, no it (still) doesn't work. I wrote the article way back when they played around with the draft and the `StackLinkedListNode`. Funny enough Microsoft has that example still on their website: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/low-level-struct-improvements#stack-based-linked-list
I created an issue for that: https://github.com/dotnet/docs/issues/32559

In the draft phase of the proposal, that should have been valid syntax for all the reasons how `ref struct` works. 
If I know more about it, I follow up.
